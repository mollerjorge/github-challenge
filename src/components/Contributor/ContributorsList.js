import React from "react";
import Contributor from "./Contributor";
import { List } from "react-virtualized";
import 'react-virtualized/styles.css'

class ContributorsList extends React.Component {
	rowRenderer = ({
		key, // Unique key within array of rows
		index, // Index of row within collection
		isScrolling, // The List is currently being scrolled
		isVisible, // This row is visible within the List (eg it is not an overscanned row)
		style // Style object to be applied to row (to position it)
	}) => {
		const content = <Contributor key={this.props.contributors[index].id}>
				{this.props.contributors[index]}
			</Contributor>;
		return (
			<div key={key} style={style}>
				{content}
			</div>
			
		);
	
	};

	render() {
		return (
			<React.Fragment>
				<List
					width={300}
					height={500}
					rowCount={this.props.contributors.length}
					rowHeight={100}
					rowRenderer={this.rowRenderer}
				/>
				
			</React.Fragment>
		);
	}
}

export default ContributorsList;
