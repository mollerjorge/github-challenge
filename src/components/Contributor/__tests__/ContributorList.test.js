import { shallow } from 'enzyme';
import React from 'react';
import ContributorsList from '../ContributorsList';

describe('ContributorsList', () => {
	const contributorsMock = [
		{
			login: 'george',
			avatar_url: 'http://mockurl.com',
			contributions: 3
		},
		{
			login: 'camille',
			avatar_url: 'http://mockurl.com',
			contributions: 5
		}
	]
	const contList = shallow(<ContributorsList contributors={contributorsMock}/>);

	it('renders correctly', () => {
		expect(contList).toMatchSnapshot();
	});

	it('displays the correct amount of Contributors', () => {
		expect(contList.find('List').dive().dive().find('Contributor').length).toEqual(2);
	});
	
});