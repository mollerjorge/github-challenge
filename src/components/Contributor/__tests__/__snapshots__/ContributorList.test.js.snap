// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributorsList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContributorsList
    contributors={
        Array [
            Object {
              "avatar_url": "http://mockurl.com",
              "contributions": 3,
              "login": "george",
            },
            Object {
              "avatar_url": "http://mockurl.com",
              "contributions": 5,
              "login": "camille",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <List
        autoHeight={false}
        estimatedRowSize={30}
        height={500}
        noRowsRenderer={[Function]}
        onRowsRendered={[Function]}
        onScroll={[Function]}
        overscanIndicesGetter={[Function]}
        overscanRowCount={10}
        rowCount={2}
        rowHeight={100}
        rowRenderer={[Function]}
        scrollToAlignment="auto"
        scrollToIndex={-1}
        style={Object {}}
        width={300}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoHeight": false,
        "estimatedRowSize": 30,
        "height": 500,
        "noRowsRenderer": [Function],
        "onRowsRendered": [Function],
        "onScroll": [Function],
        "overscanIndicesGetter": [Function],
        "overscanRowCount": 10,
        "rowCount": 2,
        "rowHeight": 100,
        "rowRenderer": [Function],
        "scrollToAlignment": "auto",
        "scrollToIndex": -1,
        "style": Object {},
        "width": 300,
      },
      "ref": Object {
        "current": null,
      },
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <List
          autoHeight={false}
          estimatedRowSize={30}
          height={500}
          noRowsRenderer={[Function]}
          onRowsRendered={[Function]}
          onScroll={[Function]}
          overscanIndicesGetter={[Function]}
          overscanRowCount={10}
          rowCount={2}
          rowHeight={100}
          rowRenderer={[Function]}
          scrollToAlignment="auto"
          scrollToIndex={-1}
          style={Object {}}
          width={300}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoHeight": false,
          "estimatedRowSize": 30,
          "height": 500,
          "noRowsRenderer": [Function],
          "onRowsRendered": [Function],
          "onScroll": [Function],
          "overscanIndicesGetter": [Function],
          "overscanRowCount": 10,
          "rowCount": 2,
          "rowHeight": 100,
          "rowRenderer": [Function],
          "scrollToAlignment": "auto",
          "scrollToIndex": -1,
          "style": Object {},
          "width": 300,
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
